### PR TITLE
Change API

### PR DESCRIPTION
**Pull request by pschicho**
This pull request focuses on removing the `UserMasses` and `UserCouplings` parameters from various files and functions, and making corresponding adjustments to the `ExportMatrixElements` function. Additionally, it includes some minor refactoring and updates to test files.

### CODE REVIEW

File: examples/xsm.m [hunk 1]
```diff
@@ -153,46 +153,35 @@
...
-FermionMass=Table[mq2,{i,1,Length[gvff[[1]]]}];
-ScalarMass=Table[ms2,{i,1,Length[gvss[[1]]]}];
-ParticleMasses={VectorMass,FermionMass,ScalarMass};
-(*
-up to the user to make sure that the same order is given in the python code
-*)
-UserMasses={mq2,mg2,mW2,mB2,ms2}; 
-UserCouplings=Variables@Normal@{Ysff,gvss,gvff,gvvv,\[Lambda]4,\[Lambda]3}//DeleteDuplicates;
+(*Light fermions*)
+LightFermions={{2,4,6,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39},"F",mq2,"LightFermions"};
```
Comment (jorindevandevis) [inline examples/xsm.m:177]: Maybe you can show the command that tells you which indices should be listed here in a comment?

File: examples/xsm.m [hunk 2]
```diff
@@ -153,46 +153,35 @@
...
 
 
 (*
 These particles do not have out-of-eq contributions
 *)
 ParticleList={ReptL,ReptR,RepGluon,RepW,RepB,RepH,RepS};
```
Comment (jorindevandevis) [inline examples/xsm.m:183]: Where did the list with lightQuarks go (repLight)? And does that mean that the other LightParticleList is just the leptons?

Comment (pschicho) [reply examples/xsm.m:183]: The RepLight was a leftover from before. Can be deleted. LightFermions are indeed all the fermions that are not top here.

File: examples/xsm.m (hunk 1 above)
Comment (pschicho) [reply examples/xsm.m]: Added in the latest update of the xsm.m example.

**Comment by og113** [comment]
I just had a look at the `CreateParticle` function. It seems like the docstring for it is mixed up. Running `?CreateParticle` currently shows:
Screenshot 2024-10-30 at 14 38 10

**Comment by og113** [comment]
Overall, I think this is a waay better than the previous API. Nice work :)

**Comment by jorindevandevis** [comment]
Did the paths change, somehow? I am now getting the error 
image
if I try to run the matrix elements files as before (from the WallGo Models folder), but also if I try to run an example in the Applications/WallGoMatrix folder

File: examples/xsm.m (hunk 1 above)
Comment (jorindevandevis) [reply examples/xsm.m]: Okay, nice! Let's also add that to the draft

**Comment by pschicho** [comment]
> Did the paths change, somehow? I am now getting the error image if I try to run the matrix elements files as before (from the WallGo Models folder), but also if I try to run an example in the Applications/WallGoMatrix folder

This should be fixed now.

**Comment by jorindevandevis** [comment]
There is a line in the idm model file at the symmetry breaking section (*using this multiple times, should not changre the outcome -> needs fixing *)
should that remain there?

**Comment by pschicho** [comment]
> There is a line in the idm model file at the symmetry breaking section (*using this multiple times, should not changre the outcome -> needs fixing *) should that remain there?

It should be removed since it was fixed in PR #34.

**Comment by og113** [comment]
For the `CreateParticle` function, the doc string for the first argument is
```
- {{r1, m1}, ..., {rn, mn}}: A list of particle parameters, where each pair {r, m} represents specific properties for each particle.
```
This seems quite unclear to me. What are these "particle parameters" exactly?

**Comment by pschicho** [comment]
> For the `CreateParticle` function, the doc string for the first argument is
> 
> ```
> - {{r1, m1}, ..., {rn, mn}}: A list of particle parameters, where each pair {r, m} represents specific properties for each particle.
> ```
> 
> This seems quite unclear to me. What are these "particle parameters" exactly? Also, for some particles there is a pair, {1,2}, and for others just a single number {1}. In e.g. the `u1_higgs_yukawa` model, none of the first arguments look like lists of pairs of parameters (they are either a single pair, or a single index).

Added some more detail in the latest formatting of docstrings.

**Review by jorindevandevis — APPROVED** [review]
I think this is a great improvement and makes the use of the code more understandable and less sensitive to errors

**Review by jorindevandevis — APPROVED** [review]

**Pull request merged**